### PR TITLE
chore(geo, storage): fix ts strict errors on geo and storage components

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
Fix ts strict errors on geo and storage components

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/components/Geo/MapView/index.tsx [hunk 1]
```diff
@@ -96,6 +96,9 @@ const MapView = forwardRef<MapRef, MapViewProps>(
         ref={ref}
         style={styleProps}
         transformRequest={transformRequest}
+        // workaround until type is fixed on react-map-gl
```
Comment (reesscot) [inline packages/react/src/components/Geo/MapView/index.tsx:99]: link to gh issue?

File: packages/react/src/components/Storage/FileUploader/UploadTracker/UploadMessage.tsx [hunk 2]
```diff
@@ -10,11 +10,11 @@ export const UploadMessage = ({
   fileState,
   errorMessage,
   percentage,
-}: FileStateProps): JSX.Element => {
+}: FileStateProps): JSX.Element | null => {
   switch (fileState) {
     case 'loading': {
       const text =
-        percentage > 0
+        percentage && percentage > 0
```
Comment (calebpollman) [inline packages/react/src/components/Storage/FileUploader/UploadTracker/UploadMessage.tsx:17]: Why is this necessary?

File: packages/react/src/components/Storage/FileUploader/UploadTracker/index.tsx [hunk 3]
```diff
@@ -90,7 +90,7 @@ export function UploadTracker({
               size="small"
               onClick={() => {
                 setTempName(name);
-                onCancelEdit();
+                onCancelEdit?.();
```
Comment (reesscot) [inline packages/react/src/components/Storage/FileUploader/UploadTracker/index.tsx:93]: shouldn't we check if it's a function first, rather than just that it's non-nullish?

File: packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx [hunk 4]
```diff
@@ -36,8 +36,9 @@ export function UploadPreviewer({
   }`;
 
   const isDisabled =
-    fileStatuses.some((status) =>
-      ['error', 'editing'].includes(status?.fileState)
+    fileStatuses.some(
+      (status) =>
+        status?.fileState && ['error', 'editing'].includes(status?.fileState)
```
Comment (calebpollman) [inline packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx:41]: Would it make sense to make `fileState` a required key of `status` rather than add this check?

File: packages/react/src/components/Storage/FileUploader/UploadTracker/UploadMessage.tsx (hunk 2 above)
Comment (sreeramsama) [reply packages/react/src/components/Storage/FileUploader/UploadTracker/UploadMessage.tsx:17]: percentage variable can be undefined

File: packages/react/src/components/Storage/FileUploader/UploadTracker/index.tsx (hunk 3 above)
Comment (sreeramsama) [reply packages/react/src/components/Storage/FileUploader/UploadTracker/index.tsx:93]: **onCancelEdit** is defined as ``onCancelEdit?: () => void;`` so only checked to make sure it is not undefined before calling it.

File: packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx (hunk 4 above)
Comment (sreeramsama) [reply packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx:41]: `fileState` a required key of `status`. Added this check as `null` is a valid value for `fileState`.

https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/components/Storage/FileUploader/types.ts#L80

File: packages/react/src/components/Geo/MapView/index.tsx (hunk 1 above)
Comment (sreeramsama) [reply packages/react/src/components/Geo/MapView/index.tsx]: Added in new commit

File: packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx (hunk 4 above)
Comment (calebpollman) [reply packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx:41]: IMO we should address the usage of `null` as a possible value of `fileState` by disallowing it (since `null` is not really a state). Granted that we will have a larger impact overall, but it will improve the overall code and feature quality

Comment (sreeramsama) [reply packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx:41]: Talked to Eric and it looks like adding another state, maybe like `idle` will be a bigger refactor so will pull this change out of this PR and have a separate PR for `FileUploader` changes.

**Review by calebpollman — APPROVED** [review]

**Review by 0618 — DISMISSED** [review]

**Review by reesscot — DISMISSED** [review]

File: packages/react/src/components/Geo/MapView/index.tsx [hunk 5]
```diff
@@ -96,6 +96,9 @@ const MapView = forwardRef<MapRef, MapViewProps>(
         ref={ref}
         style={styleProps}
         transformRequest={transformRequest}
+        // workaround until type is fixed on react-map-gl: https://github.com/visgl/react-map-gl/issues/1973
+        fog={props.fog ?? undefined}
```
Comment (reesscot) [inline packages/react/src/components/Geo/MapView/index.tsx:100]: I think the react unit tests might be failing because this need a unit test to bump coverage back to 80%

Comment (sreeramsama) [reply packages/react/src/components/Geo/MapView/index.tsx:100]: Added unit test for mapview to bump up coverage

**Review by reesscot — APPROVED** [review]

File: packages/react/src/components/Geo/MapView/__tests__/MapView.test.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+          if (new URL(url).hostname.endsWith('.amazonaws.com')) {
+            return {
+              url: 'https://maps.geo.us-east-2.amazonaws.com/maps/v0/maps/map5df169f7-staging/style-descriptor',
+            };
+          }
+        },
+      };
+    }),
+  };
+});
```
Comment (wlee221) [inline packages/react/src/components/Geo/MapView/__tests__/MapView.test.tsx:27]: Tiny nit:

```suggestion
jest.mock("maplibre-gl-js-amplify", () => ({
  AmplifyMapLibreRequest: jest.fn().mockImplementation(() => {
    return {
      amplifyTransformRequest: true,
      transformRequest: (url, resourceType) => {
        if (resourceType === "Style" && !url.includes("://")) {
          url = `https://maps.geo.us-east-2.amazonaws.com/maps/v0/maps/${url}/style-descriptor`;
        }
        if (new URL(url).hostname.endsWith(".amazonaws.com")) {
          return {
            url: "https://maps.geo.us-east-2.amazonaws.com/maps/v0/maps/map5df169f7-staging/style-descriptor",
          };
        }
      },
    };
  }),
}));
```

**Review by wlee221 — COMMENTED** [review]
One nit, but lgtm!

File: packages/react/src/components/Geo/MapView/__tests__/MapView.test.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+jest.mock('react-map-gl', () => () => {
```
Comment (wlee221) [inline packages/react/src/components/Geo/MapView/__tests__/MapView.test.tsx:29]: Can we add a comment why we need to mock this for future readers?

**Review by wlee221 — APPROVED** [review]

File: packages/react/src/components/Geo/MapView/__tests__/MapView.test.tsx (hunk 7 above)
Comment (sreeramsama) [reply packages/react/src/components/Geo/MapView/__tests__/MapView.test.tsx:28]: added comment in latest commit

**Review by reesscot — APPROVED** [review]

File: packages/react/src/components/Geo/MapView/__tests__/MapView.test.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import { Amplify, Auth } from 'aws-amplify';
+
+import { MapView } from '../index';
+
+import { act } from 'react-dom/test-utils';
```
Comment (calebpollman) [inline packages/react/src/components/Geo/MapView/__tests__/MapView.test.tsx:7]: Any reason we can't use `act` from `@testing-library/react`?

File: packages/react/src/components/Geo/MapView/__tests__/MapView.test.tsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+currentUserCredentialsSpy.mockImplementation(() =>
+  Promise.resolve(currentUserCredentials)
+);
+
+describe('Map component', () => {
+  it('should render', async () => {
+    configureSpy.mockReturnValue(partialAmplifyConfig);
+    await act(async () => {
+      render(<MapView />);
+    });
```
Comment (calebpollman) [inline packages/react/src/components/Geo/MapView/__tests__/MapView.test.tsx:58]: Are we certain we need the `act` and `async`/`await` for this test?
